File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
from math import sqrt

def solution(numbers):
    temp_set = set()
    answer = set()
    for i in range(1, len(numbers) + 1):
        for temp in permutations(numbers, i):
            temp_set.add(int("".join(temp)))

    for i in temp_set:
        if i > 1:  # 0과 1은 소수가 아님
            is_prime = True
            for j in range(2, int(sqrt(i)) + 1):
                if i % j == 0:
                    is_prime = False
                    break
            if is_prime:
                answer.add(i)
    return len(answer)

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    static Set<Integer> set = new HashSet<>();
    public int solution(String numbers) {
        recursive("", numbers);
        int cnt = 0;
        for(int i : set){
            if(isPrime(i)){
                cnt++;
            }
        }
        return cnt;
    }
    
    static void recursive(String comb, String others){
        if(!comb.equals("")){
            set.add(Integer.parseInt(comb));
        }
        for(int i = 0; i < others.length(); i++){
            recursive(comb+others.charAt(i), others.substring(0,i) + others.substring(i+1));
        }
    }
    
    static boolean isPrime(int n){
        if(n < 2){
            return false;
        }
        for(int i = 2; i <= Math.sqrt(n); i++){
            if(n % i == 0){
                return false;
            }
        }
        
        return true;
    }
}